gobblin-runtime/src/main/java/org/apache/gobblin/runtime/api/MutableSpecCatalog.java

```diff
@@ -36,5 +36,5 @@
   /**
    * Removes an existing {@link Spec} with the given URI. A no-op if such {@link Spec} does not exist.
```

Comment:
The comment says that if the spec is not found then it is a no-op. If that is not the behavior then the comment should be changed to say that it raises a `SpecNotFoundException`.

Reply:
Corrected the comment. I chose to throw an exception to avoid verifying the existence of spec before calling remove(), in order to avoid duplication of work